lib/cli/bin/generate.js

```diff
@@ -1,201 +1,50 @@
-import updateNotifier from 'update-notifier';
 import program from 'commander';
 import chalk from 'chalk';
-import detect from '../lib/detect';
-import hasYarn from '../lib/has_yarn';
-import types from '../lib/project_types';
-import { commandLog, codeLog, paddedLog, installDeps } from '../lib/helpers';
 import pkg from '../package.json';
-import angularGenerator from '../generators/ANGULAR';
-import meteorGenerator from '../generators/METEOR';
-import reactGenerator from '../generators/REACT';
-import reactNativeGenerator from '../generators/REACT_NATIVE';
-import reactNativeScriptsGenerator from '../generators/REACT_NATIVE_SCRIPTS';
-import reactScriptsGenerator from '../generators/REACT_SCRIPTS';
-import sfcVueGenerator from '../generators/SFC_VUE';
-import updateOrganisationsGenerator from '../generators/UPDATE_PACKAGE_ORGANIZATIONS';
-import vueGenerator from '../generators/VUE';
-import polymerGenerator from '../generators/POLYMER';
-import webpackReactGenerator from '../generators/WEBPACK_REACT';
-import mithrilGenerator from '../generators/MITHRIL';
-import markoGenerator from '../generators/MARKO';
-import htmlGenerator from '../generators/HTML';
-import riotGenerator from '../generators/RIOT';
+import initiate from '../lib/initiate';
+import yarnSpawnSync from '../lib/yarn_spawn_sync';
+import { codeLog } from '../lib/helpers';
 
 const logger = console;
 
+if (process.argv[1].includes('getstorybook')) {
+  logger.log(chalk.yellow('WARNING: getstorybook is deprecated.'));
+  logger.log(chalk.yellow('The official command to install Storybook from now on is:\n'));
+  codeLog(['storybook init']);
+  logger.log();
+  initiate({}, pkg);
+  process.exit(0);
+}
+
 program
-  .version(pkg.version)
+  .command('init')
+  .description('Initialize Storybook into your project.')
   .option('-f --force', 'Forcely add storybook')
   .option('-s --skip-install', 'Skip installing deps')
   .option('-N --use-npm', 'Use npm to install deps')
   .option('-p --parser <babel | babylon | flow>', 'jscodeshift parser')
   .option('-h --html', 'Add storybook for HTML')
-  .parse(process.argv);
-
-const welcomeMessage = 'getstorybook - the simplest way to add a storybook to your project.';
-logger.log(chalk.inverse(`\n ${welcomeMessage} \n`));
-
-const useYarn = Boolean(program.useNpm !== true) && hasYarn();
+  .action(options => initiate(options, pkg));
 
-const npmOptions = {
-  useYarn,
-};
-
-const runStorybookCommand = useYarn ? 'yarn run storybook' : 'npm run storybook';
-
-// Update notify code.
-updateNotifier({
-  pkg,
-  updateCheckInterval: 1000 * 60 * 60, // every hour (we could increase this later on.)
-}).notify();
+program
+  .command('start')
+  .description('Start the local Storybook server')
+  .option('-N --use-npm', 'Use NPM to start the Storybook server')
```

Comment:
When the CLI was made, I don't think NPM had the lockfile yet (i think), that's why the flag was added to the command. On the other hand, we can also argue that what should happen when both files are present (for whatever reason). Although I definitely agree with your solution if we want to prioritize NPM over Yarn, or the other way around which happens now, it would be a great idea

Reply:
also `npm-shrinkwrap.json` 😅